orttraining/orttraining/python/training/api/module.py

```diff
@@ -17,14 +17,15 @@ class Module:
 
     training: bool
 
-    def __init__(self, train_model_uri, state, eval_model_uri=None) -> None:
+    def __init__(self, train_model_uri, state, eval_model_uri=None, provider_type="cpu") -> None:
         """
         Initializes Model for Training.
         __init__ will call an internatl function to create the model.
         """
         # TODO : Add support for bytes on train_model_uri and eval_model_uri.
         self.training = True
-        self._model = C.Module(train_model_uri, state._state, eval_model_uri)
+        self._provider_type = provider_type
```

Comment:
Shall we perform validation here? what happens if the user provide `CPU` instead of `cpu`? Or `Cuda` instead of `cuda`? We could potentially benefit from having a class called `ProviderType`, that does the validation and stores anything device related. You could use the existing `C.OrtDevice`.
Please add necessary documentation for the class and expected arguments and return values.

Look at the ortmodule does it: https://github.com/microsoft/onnxruntime/blob/main/orttraining/orttraining/python/training/ortmodule/ortmodule.py#L27-L35 and try to see if we can have at least that amount of detail here.

I think we might benefit from having a

Reply:
We can add a validation and add return an error that we are expecting, cuda or cpu as device options if users put a different value,
why would we need to add a new class for this? we can handle the initialization of the execution providers in the c++ side, this way we can keep the python bindings simple for the users, and they don't have to deal with all these configurations.

Reply:
for example, if users enter the device as "cuda:0" or "cuda:1", the class can be responsible for storing the device as cuda and the index as 0 or 1 and so on. Trying to replicate `torch.device`.

Reply:
this class will be on python side or on C++ side and we expose it as python binding ?

Reply:
This can be an internal class if needed. So, users can pass in a string, and we can internally parse the string and store it in `C.OrtDevice` as a member variable of `Module`.